Server/src/services/getDataService.ts

```diff
@@ -0,0 +1,110 @@
+import { IDatasetResponseModel } from "../models/interfaces/DatasetResponseModelInterface";
+import { IDataRequestModel } from "../models/interfaces/DataRequestModelInterface";
+import { DataQueryModel } from "../models/DataQueryModel";
+
+export class retrieveData {
+    private dataQuery: DataQueryModel;
+
+    constructor() {
+        this.dataQuery = new DataQueryModel();
+    }
+
+    async getArrayOfDatasets(receivedData: IDataRequestModel) {
+        let datasetReceived = receivedData.datasetId;
+        let materialReceived = receivedData.material;
+        let firstNameReceived = receivedData.firstName;
+        let lastNameReceived = receivedData.lastName;
+        let yearReceived = receivedData.year;
+        let categoryReceived = receivedData.categoryId;
+        let subcategoryReceived = receivedData.subcategoryId;
+        let selectedDatasetIds = [];
+        if (datasetReceived != undefined) {
```

Comment:
This can be removed with the selectedDatasetIds already set, and if for undefined (changes we implemented during our chat)

Reply:
Done in commit 6a0eac9